aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1602,6 +1602,68 @@ Tensor& linalg_norm_out(Tensor& result, const Tensor& self, std::string ord, opt
   return linalg_norm_out_impl(result, self, c10::nullopt, ord, opt_dim, keepdim, opt_dtype);
 }
 
+Tensor linalg_tensorinv(const Tensor& self, int64_t ind) {
+  /*
+  The idea is to reduce the problem to 2D square matrix inversion.
+  Step 1. calculate the shape of the result and the shape of the intermediate 2D matrix.
+  Step 2. reshape `self` to 2D matrix.
+  Step 3. invert the 2D matrix self.to_2D()
+          there is no quick way to find out whether the matrix is invertible,
+          so at this stage an error from at::inverse can be thrown
```

Comment:
This is a very helpful description. How does this error checking work on CUDA? Does it cause cross-device memory movement? (However it works, a brief statement should be added here explaining it.)

Reply:
So `at::inverse` calls to cusolver's `getrf` & `getrs`. Error checking is done via checking the `devInfo` integer ([cusolver docs link](https://docs.nvidia.com/cuda/cusolver/index.html#cuSolverDN-lt-t-gt-getrf)), if it is larger than 0, then there is a problem. Since it lies in the GPU memory, checking the error causes cross-device memory movement. That's an expensive thing to do. For example, CuPy has decided not to check for errors by default to avoid a performance drop.

Reply:
I also think we should consider an alternative method to check for errors. cc @ngimel who's thought about this issue, too. 

As long as we mention the current behavior here (relying on at::inverse, which syncs) I think we can keep the current behavior.